[docs] Clarify how app-specific deep links work with dev builds (#25970)
expo/expo: docs/pages/develop/development-builds/development-workflows.mdx

{/* TODO: @aman move this section out of this page -- make it part of the main path or a standalone guide, this is important part that should be searchable for eg: Deep linking URLs for Development builds/EAS Builds (when using expo-dev-client) */}

> ### Deep linking to an update's URL

Reviewer: What does update signify here? I don't think we signify it in this section.

Author: IMO "Deep linking URL's" could mean _any_ deep link, but the original text is only about a very specific deep link that opens an update URL.

Of course, I kind of break this by then putting info about deep links in general as a sub-heading under this. I could move that subsection up a level.

Reviewer: Thanks for the clarifixation. It has been a while since I've looked at this doc, I was lacking context. I think the title and the sub section are both fine.

In the example above, the `scheme` is `exp+app-slug`, and the `manifestUrl` is a project with an ID of `F767ADF57-B487-4D8F-9522-85549C39F43F` and a channel of `main`.

#### App-specific deep links

> When testing deep links in your development build, such as when navigating to a specific screen in an Expo Router app, or testing redirecting back to your app during an Oauth login flow, construct the URL exactly as you would as if you were deep-linking into a standalone build of your app (e.g., `myscheme://path/to/screen`).

Reviewer: ```suggestion
When testing deep links in your development build, such as when navigating to a specific screen in an Expo Router app or testing redirecting back to your app during an Oauth login flow, construct the URL exactly as you would if you were deep-linking into a standalone build of your app (for example, `myscheme://path/to/screen`).
```

> Your project must be already open in the development build in order for an app-specific deep link to work. Cold-launching a development build with an app-specific deep link is not currently supported. In your app-specific deep links, avoid using `expo-development-client` in the path, as this is a reseved path used for launching an updates URL.

Reviewer: ```suggestion
Your project must be already open in the development build for an app-specific deep link to work. Cold-launching a development build with an app-specific deep link is not currently supported. Avoid using `expo-development-client` in your app-specific deep links in the path, as it is a reserved path used for launching an updated URL.
```
